ext/eventcart/eventcart.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+require_once 'eventcart.civix.php';
+
+/**
+ * Implements hook_civicrm_config().
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_config/
+ */
+function eventcart_civicrm_config(&$config) {
+  if (isset(Civi::$statics[__FUNCTION__])) {
+    return;
+  }
+  Civi::$statics[__FUNCTION__] = 1;
+  Civi::dispatcher()->addListener('hook_civicrm_pageRun', 'CRM_Event_Cart_PageCallback::run');
+
+  _eventcart_civix_civicrm_config($config);
+
+  // If upgrading from an older version of CiviCRM (ie. when eventcart was part of core)
+  //   the eventcart extension will have already been "installed".
+  // If installing a new instance of CiviCRM the installer will never run because we currently
+  //   add the enabled extension directly to the database.
+  // The _config hook is one of the earliest hooks to run so we trigger a check here that will work
+  //   in all circumstances that the extension could be installed but not.
+  //   If we are missing the civicrm_event_carts table we trigger the installer.
+  // @fixme This is a hack to make sure we run the installer on a core extension
+  if (!CRM_Core_DAO::checkTableExists('civicrm_event_carts')) {
+    $upgrader = _eventcart_civix_upgrader();
+    // @fixme As a core extension the wrong path is generated by CRM_Event_Cart_Upgrader_Base::instance()
+    //   Replacing realpath with E::path() fixes it here and we should consider updating civix to do the same.
+    $upgrader->onInstall();
+    if (is_callable([$upgrader, 'onPostInstall'])) {
+      $upgrader->onPostInstall();
+    }
+  }
+}
```

Comment:
@eileenmcnaughton @mlutfy @colemanw I've added an "install" process here, directly in the extension. It's a bit of a hack but also good because it doesn't require changes to CiviCRM core at this stage.